src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -78,41 +92,257 @@ static DamageTable D_8099F4E8 = {
+void EnZo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    SkelAnime* skelAnime = &this->skelAnime;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EA60.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D208, NULL, this->jointTable, this->morphTable, 20);
+    EnZo_SetAnimation(skelAnime, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->path = func_8013D648(globalCtx, ENZO_GET_PATH(&this->actor), 0x3F);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnZo_Walk;
+    this->actor.gravity = -4.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EBD8.s")
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EC50.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099ED4C.s")
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EE24.s")
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnZo_LookAtPlayer(this, globalCtx);
+    EnZo_PlayWalkingSound(this, globalCtx);
+    EnZo_UpdateCollider(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Init.s")
+s32 EnZo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                          Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Destroy.s")
+    if (limbIndex == 15) {
+        Matrix_InsertTranslation(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->headRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->headRot.x, MTXMODE_APPLY);
+        Matrix_InsertTranslation(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+    if (limbIndex == 8) {
+        Matrix_InsertXRotation_s(-this->upperBodyRot.y, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(-this->upperBodyRot.x, MTXMODE_APPLY);
+    }
+    if ((limbIndex == 8) || (limbIndex == 9) || (limbIndex == 12)) {
+        rot->y += (s16)(Math_SinS(this->limbRotY[limbIndex]) * 200.0f);
+        rot->z += (s16)(Math_CosS(this->limbRotZ[limbIndex]) * 200.0f);
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Update.s")
+void EnZo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnZo* this = (EnZo*)actor;
+    Vec3f sp30 = D_8099F5CC;
+    Vec3f zeroVec = D_8099F5D8;
```

Comment:
This looks like those should be inlined
```suggestion
    Vec3f sp30 = { 400.0f, 0.0f, 0.0f };
    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
```

You'll also need to remove `D_8099F5CC` and `D_8099F5D8` from the top of the file and push `D_8099F5E8`  down this function so the data order is preserved